Library/Homebrew/extend/os/mac/hardware.rb

```diff
@@ -24,4 +24,20 @@ def self.oldest_cpu(version = nil)
       generic_oldest_cpu
     end
   end
+
+  # Override
+  # Mirrors version-dependent logic of oldest_cpu
+  sig { params(version: T.nilable(Version)).returns(String) }
+  def self.rustflags_target_cpu(version = nil)
+    version = if version
+      MacOSVersion.new(version.to_s)
+    else
+      MacOS.version
+    end
+    if Hardware::CPU.intel? && version >= :mojave
+      "'-Ctarget-cpu=nehalem'"
```

Comment:
```suggestion
      "--codegen target-cpu=nehalem"
```
1. Let callers handle the quoting, if it's needed.
2. Use the long version of the flag.
3. I'd still rather use `oldest_cpu` here instead of hard-coding `nehalem`.